libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -276,6 +278,14 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 	if (_variable.annotation().type)
 		return;
 
+	if (auto typeName = dynamic_cast<UserDefinedTypeName const*>(&_variable.typeName()))
+	   if (dynamic_cast<ModifierDefinition const*>(typeName->annotation().referencedDeclaration))
```

Comment:
We might also want to move this down to where `typeName` is accessed.

Reply:
If we move this down we get assert failures because the location info is not available but accessed a few lines down.
With list you mean we should name the things a variable can be instead of that it can't be a modifier?